src/overlays/actors/ovl_En_Yb/z_en_yb.c

```diff
@@ -37,59 +44,443 @@ const ActorInit En_Yb_InitVars = {
+        this->cutscenes[i] = tempCutscene;
+        if (tempCutscene != -1) {
+            this->actor.cutscene = tempCutscene;
+            tempCutscene = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Yb/func_80BFA9D4.s")
+    this->cutsceneIndex = -1;
+    this->actor.cutscene = this->cutscenes[0];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Yb/func_80BFAB4C.s")
+    // between midnight and morning start spawned
+    if (gSaveContext.time < CLOCK_TIME(6, 0)) {
+        this->alpha = 255;
+    } else { // else (night 6pm to midnight): wait to spawn
+        this->alpha = 0;
+        this->actionFunc = EnYb_WaitForMidnight;
+        this->actor.flags &= ~ACTOR_FLAG_1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Yb/func_80BFABF0.s")
+    // check if already healed
+    if (gSaveContext.weekEventReg[82] & 4) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Yb/func_80BFAC88.s")
+void EnYb_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnYb* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Yb/func_80BFAE80.s")
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Yb/EnYb_Update.s")
+void func_80BFA2FC(GlobalContext* globalCtx) {
+    if (INV_CONTENT(ITEM_MASK_KAMARO) == ITEM_MASK_KAMARO) {
+        func_80151BB4(globalCtx, 0x34);
+    }
+    func_80151BB4(globalCtx, 0xF);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Yb/func_80BFB074.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Yb/func_80BFB0E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Yb/EnYb_Draw.s")
+/**
+ * custom shadow draw function of type ActorShadowFunc
+ */
+void EnYb_ActorShadowFunc(Actor* thisx, Lights* mapper, GlobalContext* globalCtx) {
+    Vec3f oldPos;
+    EnYb* this = THIS;
+
+    if (this->alpha > 0) {
+        if (this->currentAnimIndex == 2) {
+            // regalloc without temp
+            f32 tempScale = (((27.0f - this->shadowPos.y) + this->actor.world.pos.y) * ((1 / 2.25f) * 0.001f)) + 0.01f;
+            this->actor.scale.x = tempScale;
+        }
+        Math_Vec3f_Copy(&oldPos, &this->actor.world.pos);
+        Math_Vec3f_Copy(&this->actor.world.pos, &this->shadowPos);
+        func_800B4AEC(globalCtx, &this->actor, 50.0f);
+
+        if (oldPos.y < this->actor.floorHeight) {
+            this->actor.world.pos.y = this->actor.floorHeight;
+        } else {
+            this->actor.world.pos.y = oldPos.y;
+        }
+
+        ActorShadow_DrawCircle(&this->actor, mapper, globalCtx);
+        Math_Vec3f_Copy(&this->actor.world.pos, &oldPos);
+        this->actor.scale.x = 0.01f;
+    }
+}
+
+/**
+ * weird animation changing function
+ * only gets called from init with static variables though: (animIndex: 2, animMode: 0, transitionRate: 0.0f)
```

Comment:
Hmm... I want to keep the comments inside of the function as a local reminder when looking at the actual code, in case someone finds the code without looking at the function description, its more organic. If I can't have both can I keep the in-function and remove the function description?

Reply:
I suppose it doesn't matter too much.